### PR TITLE
Return detailed error if http request was failed

### PR DESCRIPTION
**Pull request by TheZeroSlave**

### CODE REVIEW

**Comment by fiorix** [comment]
Not nitpicking but please s/Http/HTTP before I can merge.

**Comment by fiorix** [comment]
ping @TheZeroSlave

File: soap/client.go [hunk 1]
```diff
@@ -185,6 +189,17 @@ func (c *Client) RoundTripSoap12(action string, in, out Message) error {
 	return doRoundTrip(c, headerFunc, in, out)
 }
 
+//Detailed soap http error
+type HttpError struct {
```
Comment (adriantam) [inline soap/client.go]: Please ensure it conforms to golint by having comment begins with the name of the element it describes.  Thanks!

**Comment by TheZeroSlave** [comment]
sorry guys, i was busy :)

**Comment by fiorix** [comment]
👍

**Pull request merged**